vignettes/benchmarks.Rmd.orig

```diff
@@ -0,0 +1,914 @@
+  runtimes_dt_long,
+  model_components,
+  by = "model"
+)
+
+# snapshot dates dictionary
+snapshot_date_labels <- c(growth = "2020-05-15", peak = "2020-06-21", decline = "2020-06-28")
+
+# Replace snapshot_date based on the dictionary
+runtimes_dt_detailed[, epidemic_phase := names(snapshot_date_labels)[match(snapshot_date, snapshot_date_labels)]]
+
+# Add model descriptions
+runtimes_dt_detailed <- merge(
+  runtimes_dt_detailed,
+  model_descriptions,
+  by = "model"
+)
+
+# Re-categorise the factor levels
+runtimes_dt_detailed <- add_fit_type(runtimes_dt_detailed)
+
+# Make all columns except timing a factor
+runtimes_dt_detailed <- make_cols_factors(runtimes_dt_detailed, except = "timing")
+
+# Add epidemic_phase factor levels to c("growth", "peak", "decline"))
+runtimes_dt_detailed <- add_epidemic_phase_levels(runtimes_dt_detailed)
+
+# Plot the timing
+timing_plot <- ggplot(data = runtimes_dt_detailed) +
+  geom_point(aes(x = epidemic_phase,
+                 y = timing,
+                 shape = model_basename,
+                 color = fitting
+                 ),
+             position = position_dodge(width = 0.4),
+             size = 3,
+             stroke = 1.2
+  ) +
+  labs(x = "Epidemic phase",
+       y = "Runtime (secs)",
+       shape = "Model",
+       fill = "Fitting method",
+       title = "Runtimes per model and fitting method",
+       caption = "Where a model is not shown, it means it failed to run"
+  ) +
+  theme_minimal() +
+  scale_color_brewer(palette = "Dark2") +
+  scale_shape_manual(values = c(1, 0, 6, 8)) +
+  facet_wrap(~fit_type, scales = "free_y", nrow = 2, strip.position = "top") +
+  theme(plot.title = element_text(size = 18),
+        strip.text = element_text(size = 13),
+        axis.title = element_text(size = 13),
+        axis.text = element_text(size = 11)
+        )
+timing_plot
+```
+
+We can see that across the board, the non-mechanistic model and non-residuals models were the fastest whereas the default model was among the slowest models for all data scenarios.
+
+### Evaluating model performance
+
+Now, we will evaluate the performance of the models using the [continuous ranked probability score (CRPS)](https://en.wikipedia.org/wiki/Scoring_rule#Continuous_ranked_probability_score). The CRPS is a [proper scoring rule](https://en.wikipedia.org/wiki/Scoring_rule#Propriety_and_consistency) that measures the accuracy of probabilistic forecasts. When comparing models, the smaller the CRPS, the better.
+
+We will focus on evaluating the results of fitting with MCMC. We will also show the results for the approximate sampling methods but will not delve deeply into them as we do not recommended for full inference due to their experimental nature currently, which makes them unstable. In a latter section, we will provide ideas on how the approximate methods can be used in `{EpiNow2}` workflows.
+
+```{r process-crps}
+# Process CRPS for Rt
+rt_crps <- process_crps(results, "R", R)
```

Comment:
I wonder if it would be good to look at the estimate on the day of the inference for a real-time measure and then e.g. the 7-day forecast rather than the whole "estimate from partial data" / "forecast" time series. Debatable perhaps but if you use the whole series I think it would make more sense to use the multivariate CRPS / energy score (which is not yet implemented in `scoringutils` but is available in `scoringRules::es_sample()`.

Reply:
Current version how computes the total CRPS over the forecast horizon, the Rt estimate on the day of the inference for a real-time measure, and then the 7-day forecast of infections.